test/e2e/connection-alice-to-bob.spec.ts

```diff
@@ -0,0 +1,218 @@
+import { expect, test } from '@playwright/test'
+import { checkEmails, extractInvite, extractPin, findNewAdminEmail } from './helpers/smtpEmails'
+
+test.describe('Connection from Alice to Bob', () => {
+  let context: any
+  let page: any
+  let adminEmailId: string
+  let invite: string | null
+  let pinForBob: string | null
+  let pinForAlice: string | null
+
+  // Create and share context
+  test.beforeAll(async ({ browser }) => {
+    context = await browser.newContext()
+  })
+
+  test.beforeEach(async () => {
+    page = await context.newPage()
+  })
+
+  test.afterEach(async () => {
+    await page.close()
+  })
+  // End-to-end process: Alice registers, invites Bob, Bob submits invite & pin, Alice submits pin
+  test('Connection from Alice to Bob', async () => {
+    test.setTimeout(10000)
+
+    await test.step('Alice registers on the platform', async () => {
+      await page.goto('http://localhost:3000')
+      const url = page.url()
+      expect(url).toContain(
+        'http://localhost:3080/realms/veritable/protocol/openid-connect/auth?response_type=code&client_id=veritable-ui&redirect_uri=http'
+      )
+
+      await page.waitForSelector('a[href*="/realms/veritable/login-actions/registration"]')
+      await page.click('a[href*="/realms/veritable/login-actions/registration"]')
+      await page.waitForURL('**/realms/veritable/login-actions/registration**')
+
+      await page.fill('#username', 'name')
+      await page.fill('#password', 'password')
+      await page.fill('#password-confirm', 'password')
+      await page.fill('#email', 'email@testmail.com')
+      await page.fill('#firstName', 'name')
+      await page.fill('#lastName', 'lastname')
+      await page.click('input[type="submit"][value="Register"]')
+      await page.waitForURL('http://localhost:3000')
```

Comment:
maybe register could be abstracted as well since this is something we will need for each suite? to be fair not a major so this does not block me from approving